htdocs/api/v0.0.1/APIBase.php

```diff
@@ -0,0 +1,251 @@
+<?php
+/**
+ * Base class to handle requests to the Loris API and perform
+ * validation common to all API requests.
+ *
+ * PHP Version 5.5+
+ *
+ * @category Main
+ * @package  API
+ * @author   Dave MacFarlane <david.macfarlane2@mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+namespace Loris\API;
+require_once __DIR__ . '/SafeExitException.php';
+require_once __DIR__ . '/../../../vendor/autoload.php';
+
+/**
+ * Base class to handle requests to the Loris API and perform
+ * validation common to all API requests.
+ *
+ * @category Main
+ * @package  API
+ * @author   Dave MacFarlane <david.macfarlane2@mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+abstract class APIBase
+{
+    var $DB;
+    var $client;
+    var $JSON;
+    var $AllowedMethods = [];
+    var $AutoHandleRequestDelegation = true;
+    var $HTTPMethod;
+
+    var $Factory;
+    var $Headers;
+
+    /**
+     * Constructor to handle basic validation
+     *
+     * @param string $method The HTTP request method
+     */
+    function __construct($method)
+    {
+        if (empty($this->AllowedMethods)) {
+            $this->AllowedMethods = ['GET'];
+        }
+        // Verify that method is allowed for this type of request.
+        if (!in_array($method, $this->AllowedMethods)) {
+            $this->header("HTTP/1.1 405 Method Not Allowed");
+            $this->header("Allow: " . join(", ", $this->AllowedMethods));
+            $this->safeExit(0);
+        }
+
+        $this->HTTPMethod = $method;
+
+        //Load config file and ensure paths are correct
+        set_include_path(
+            get_include_path() . ":" .
+            __DIR__ . "/../../../php/libraries"
+        );
+        include_once 'NDB_Client.class.inc';
+
+        $this->Factory = \NDB_Factory::singleton();
+        $this->client  = new \NDB_Client();
+        // Even though it's not a command line client, this prevents
+        // the login related to showing the login screen from applying,
+        // then we manually
+        $this->client->makeCommandLine();
```

Comment:
By setting this before initializing the SinglePointLogin flag doesn't get set so the $this->client->isLoggedIn() function call returns false